src/generators/typescript/index.js

```diff
@@ -43,22 +43,34 @@ class TypeScript extends Generator {
   async renderIndex(data) {
     this.indexTemplate =
       this.indexTemplate ||
-      (await this.loadTemplate(__dirname + "/index.js.mustache"));
+      (await this.loadTemplate(__dirname + "/index.ts.mustache"));
 
     return this.indexTemplate(data);
   }
 
+  /**
+   * @param {string} typeName
+   */
+  filterModel(typeName, model) {
+    /* EventStatusType is erroneously made into a model as well as an enum. Because the TS generator puts enums and
+    models into the same directory, this causes issues */
+    if (typeName === 'schema:EventStatusType') {
+      return false;
+    }
+    return true;
+  }
```

Comment:
This is unfortunately necessary as it seems that models-lib produces a schema EventStatusType model (which is empty) as well as the enum. This seems like a bug to me. In this code, I've just filtered out the model

You can see an example of this in the existing models-ts project: https://github.com/openactive/models-ts/blob/main/src/schema/models/EventStatusType.js vs https://github.com/openactive/models-ts/blob/main/src/schema/enums/EventStatusType.js

Reply:
This is a very strange bug - doesn't seem to relate to the schema.org model either!